pytorch_lightning/loops/fit_loop.py

```diff
@@ -195,6 +197,22 @@ def on_run_start(self) -> None:  # type: ignore[override]
         """Calls the ``on_train_start`` hook."""
         # reset train dataloader and val dataloader
         self.trainer.reset_train_val_dataloaders(self.trainer.lightning_module)
+
+        # global_step is incremented during checkpointing (#11555)
+        if self.restarting and self.trainer.num_training_batches not in (0, float("inf")):
+            self.trainer.accumulate_grad_batches = self.trainer.accumulation_scheduler.get_accumulate_grad_batches(
```

Comment:
Don't we need to set this attribute unconditionally?

Reply:
sorry, didn't get you.

This is updated at every epoch. Even if I remove this line it will still work, but just kept it to be extra sure.

Reply:
I mean if we should do

```python
self.trainer.acccumulate_grad_batches = ...
if self.restarting and ...:
    ...
```

Reply:
we need the value under certain conditions. if the condition doesn't satisfy, computing it outside won't do any good.